File: backend/app/schemas/library.py
from pydantic import BaseModel

from app.models.content import ContentType
from app.schemas.album import AlbumListResponse
from app.schemas.content import ContentResponse


class LibraryResponse(BaseModel):
    enrolled_albums: list[AlbumListResponse]
    saved_snippets: list[ContentResponse]


class ContentSearchResult(BaseModel):
    content_id: int
    title: str
    content_type: ContentType
    album_title: str | None
    similarity_score: float
    is_saved: bool


class MentorRequest(BaseModel):
    message: str


class MentorSource(BaseModel):
    content_id: int
    title: str


class MentorResponse(BaseModel):
    reply: str
    sources: list[MentorSource]

File: frontend/src/lib/api/library.ts
import { apiFetch } from './client';
import type { AlbumListResponse, ContentListResponse } from './types';

export interface LibraryResponse {
  enrolled_albums: AlbumListResponse[];
  saved_snippets: ContentListResponse[];
}

export interface ContentSearchResult {
  id: number;
  title: string;
  body: string;
  content_type: string;
}

export interface MentorResponse {
  answer: string;
  sources: ContentSearchResult[];
}

export async function getLibrary(): Promise<LibraryResponse> {
  return apiFetch<LibraryResponse>('/library/');
}

export async function searchLibrary(query: string): Promise<ContentSearchResult[]> {
  return apiFetch<ContentSearchResult[]>(`/library/search?q=${encodeURIComponent(query)}`);
}

export async function mentorQuery(query: string): Promise<MentorResponse> {
  return apiFetch<MentorResponse>('/library/mentor', {
    method: 'POST',
    body: JSON.stringify({ query }),
  });
}

export async function saveSnippet(contentId: number): Promise<void> {
  await apiFetch<void>(`/library/snippets/${contentId}`, { method: 'POST' });
}

export async function unsaveSnippet(contentId: number): Promise<void> {
  await apiFetch<void>(`/library/snippets/${contentId}`, { method: 'DELETE' });
}
